ideal_response/ideal_response_functions.R

```diff
@@ -1,189 +1,261 @@
+  ideal_response_1_s <- thicken(
+    ideal_response_1_s,
+    paste(duration, "s"),
+    colname = "time_group",
+    rounding = "up",
+    start_val = offset - as.numeric(duration)
+  )
+  ideal_response_1_s <- thicken(
+    ideal_response_1_s,
+    paste(duration, "s"),
+    colname = "time_group2",
+    rounding = "down",
+    by = "ts",
+    start_val = offset - as.numeric(duration)
+  )
+  ideal_response_1_s[ideal_response_1_s$ts == ideal_response_1_s$time_group2,]$time_group <-
+    ideal_response_1_s[ideal_response_1_s$ts == ideal_response_1_s$time_group2,]$time_group2
   ideal_response_1_s <- filter(
-    ideal_response_1_s, 
-    ts<=max(ideal_response_1_s[ideal_response_1_s$ts==ideal_response_1_s$time_group2,]$time_group2))
+    ideal_response_1_s,
+    ts <= max(ideal_response_1_s[ideal_response_1_s$ts == ideal_response_1_s$time_group2,]$time_group2)
+  )
   ideal_response_1_s <- filter(
-    ideal_response_1_s, 
-    ts>=min(ideal_response_1_s[ideal_response_1_s$ts==(ideal_response_1_s$time_group2+1),]$ts))
+    ideal_response_1_s,
+    ts >= min(ideal_response_1_s[ideal_response_1_s$ts == (ideal_response_1_s$time_group2 + 1),]$ts)
+  )
   ideal_response_downsampled <- group_by(ideal_response_1_s, time_group)
-  ideal_response_downsampled <- summarise(ideal_response_downsampled, f=last(f), norm_power=mean(norm_power))
+  ideal_response_downsampled <- summarise(ideal_response_downsampled, f = last(f), norm_power = mean(norm_power))
   ideal_response_downsampled <- data.frame(ideal_response_downsampled)
   return(ideal_response_downsampled)
 }
 
-calc_error_metric_and_compliance <- function(combined_data, ideal_response_downsampled){
+calc_error_metric_and_compliance <- function(combined_data, ideal_response_downsampled) {
   error_by_c_id <- calc_error_metric(combined_data, ideal_response_downsampled)
   threshold_error <- calc_threshold_error(ideal_response_downsampled)
   error_by_c_id <- calc_compliance_status(error_by_c_id, threshold_error)
-  combined_data <- left_join(combined_data, error_by_c_id, by="site_id")
+  combined_data <- left_join(combined_data, error_by_c_id, by = c("site_id"))
   return(combined_data)
 }
 
-calc_error_metric <- function(combined_data, ideal_response_downsampled){
-  combined_data <- distinct(combined_data, site_id, ts, .keep_all=TRUE)
-  combined_data <- inner_join(combined_data, ideal_response_downsampled, by=c("ts"="time_group"))
-  combined_data <- mutate(combined_data, 
-                          abs_percent_diff_actual_cf_ideal=abs(Site_Event_Normalised_Power_kW-norm_power)/norm_power)
-  combined_data <- mutate(combined_data, percent_diff_actual_cf_ideal=(Site_Event_Normalised_Power_kW-norm_power)/norm_power)
+calc_error_metric <- function(combined_data, ideal_response_downsampled) {
+  combined_data <- distinct(combined_data, site_id, ts, .keep_all = TRUE)
+  combined_data <- inner_join(combined_data, ideal_response_downsampled, by = c("ts", "time_group"))
```

Comment:
Original "ts"="time_group" has become "ts", "time_group" and causes error

Reply:
Thank you! I'd accidentally done this for a bunch of columns and realised the mistake and tried to revert it, but evidently I'd missed a few cases